### PR TITLE
add list of goods hook task solution

### PR DESCRIPTION
**Pull request by opiquem**
[DEMO LINK](https://maksym-bobryk.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,47 +16,114 @@ export const goodsFromServer = [
...
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, reverse] = useState(false);
+  const [sortType, sort] = useState(SortType.NONE);
```
Comment (dinaminahi) [inline src/App.tsx:58]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
  const [sortType, seSortType] = useState(SortType.NONE);
```

File: src/App.tsx [hunk 2]
```diff
@@ -15,47 +16,114 @@ export const goodsFromServer = [
...
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {resetFlag && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => reset()}
```
Comment (dinaminahi) [inline src/App.tsx:113]:
```suggestion
            onClick={reset}
```

File: src/App.tsx [hunk 3]
```diff
@@ -15,47 +16,114 @@ export const goodsFromServer = [
...
+  const handleReverseToggle = () => {
+    reverse((current) => !current);
+  };
+
+  const reset = () => {
+    reverse(false);
+    sort(SortType.NONE);
+  };
+
+  const visibleGoods = getReorderedGoods(
```
Comment (dinaminahi) [inline src/App.tsx:69]:
```suggestion
  const reorderedGoods = getReorderedGoods(
```

File: src/App.tsx [hunk 4]
```diff
@@ -15,47 +16,114 @@ export const goodsFromServer = [
...
+    goodsFromServer,
+    { isReversed, sortType },
+  );
+
+  const resetFlag = sortType !== SortType.NONE || isReversed;
```
Comment (dinaminahi) [inline src/App.tsx:74]:
```suggestion
  const isResetVisible = sortType !== SortType.NONE || isReversed;
```

**Review by dinaminahi — CHANGES_REQUESTED** [review]
Good job 👍 
Few things to fix

**Review by 33Orange — APPROVED** [review]
Well done

**Pull request closed without merging**